src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -146,15 +151,24 @@ void ResourceGatheringOperation::rgo_tick() {
 		owner_pops_cache,
 		total_owner_count_in_state_cache
 	);
-
-	revenue_yesterday = output_quantity_yesterday * production_type.get_output_good().get_base_price(); //TODO sell on market
-
-	pay_employees(
-		revenue_yesterday,
-		total_worker_count_in_province,
-		owner_pops_cache,
-		total_owner_count_in_state_cache
-	);	
+	market_instance.place_market_sell_order({
+		production_type.get_output_good(),
+		output_quantity_yesterday,
+		[
+			this,
+			total_worker_count_in_province,
+			owner_pops_cache = std::move(owner_pops_cache),
+			total_owner_count_in_state_cache
+		](const SellResult sell_result) mutable -> void {
```

Comment:
```suggestion
			owner_pops_cache,
			total_owner_count_in_state_cache
		](const SellResult sell_result) -> void {
```
This doesn't need to be `mutable` and `owner_pops_cache` doesn't need to be moved, it's a pointer not a vector itself so moving just copies it anyway (you couldn't move the value anyway even with `std::move(*owner_pops_cache)` as it's a `const*` to a `State`'s pop cache, and if it was made a non-const pointer then moving it would clear the `State`'s pop cache until the next gamestate update)

Reply:
Nice catch. It's a leftover from earlier code. Should've been cleaned up in [the state owner cache PR](https://github.com/OpenVicProject/OpenVic-Simulation/pull/224).

Also noticed `produce` doesn't need the cache and so I removed it. On top of that, I renamed it to `owner_pops_cache_nullable` for clarity.